### PR TITLE
feat(conductor, composer, sequencer, relayer): setup basics for metrics

### PR DESCRIPTION
**Pull request by joroshiba**
## Summary
Adds the setup for metrics into core telemetry init, adds config for it to all our standard services.

## Background
Metrics are currently only available through cometbft or rollup nodes, we want to have metrics in our configuration

## Changes
- Updates all service configs
- Updates telemetry to enable initializing prometheus metrics
- Adds a couple of metrics for sequencer-relayer

## Breaking Changelist
- new configuration fields

## Related Issues
Link any issues that are related, prefer full github links.

closes

### CODE REVIEW

File: crates/astria-sequencer-relayer/src/relayer.rs [hunk 1]
```diff
@@ -164,6 +168,7 @@ impl Relayer {
         // Then report update the internal state or report if submission failed
         match submission_result {
             Ok(height) => self.state_tx.send_modify(|state| {
+                counter!("successful_submission_celestia_height", height);
```
Comment (noot) [inline crates/astria-sequencer-relayer/src/relayer.rs:171]: what does this do if metrics are not enabled? nothing?

**Review by noot — APPROVED** [review]

File: crates/astria-sequencer-relayer/src/relayer.rs (hunk 1 above)
Comment (joroshiba) [reply crates/astria-sequencer-relayer/src/relayer.rs:171]: Yeah basically, without an exporter there's nowhere for them to go. We can also do this in libraries to implement metrics in them, if imported and exporter attached get the metrics if not nothing.

**Review by SuperFluffy — APPROVED** [review]
I approve of this PR in general, but there are a few things I would ask you to fix before merging:

1. error handling for setting up metrics since the telemetry crate handles its errors
2. delayed/lazy parsing of the prometheus addr into SocketAddr (so that one need not set it if metrics are disabled)
3. the `metrics` modules together with the `use ::metrics` imports and `use crate::metrics as metric_labels` imports feel very clunky. I see now why penumbra chose to re-export metrics into the modules.

File: crates/astria-celestia-client/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    register_gauge!(ROLLUP_BLOBS_PER_CELESTIA_TX, "library" => "astria_celestia_client");
+    describe_gauge!(
+        ROLLUP_BLOBS_PER_CELESTIA_TX,
+        Unit::Count,
+        "The total number of rollup blobs included in the last Celestia submission"
+    );
+}
+
+pub const ROLLUP_BLOBS_PER_ASTRIA_BLOCK: &str =
+    "astria_celestia_client_rollups_blobs_per_astria_block";
```
[the remark this answers is not in the document]
Comment (joroshiba) [reply crates/astria-celestia-client/src/metrics.rs]: will this use the crate name of the target or of the library it's in?

[the remark this answers is not in the document]
Comment (SuperFluffy) [reply crates/astria-celestia-client/src/metrics.rs]: Did an experiment:
```rust
// $ cat foo/src/librs
const FOO: &str = concat!(env!("CARGO_CRATE_NAME"), "_foo");

pub fn foo() {
    println!("{FOO}");
}

// $ cat bar/src/main.rs
const BAR: &str = concat!(env!("CARGO_CRATE_NAME"), "_bar");

pub fn bar() {
    println!("{BAR}");
}

fn main() {
    foo::foo();
    bar();
}
```
Results in:
```sh
$ cargo run --bin bar
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
     Running `target/debug/bar`
foo_foo
bar_bar
```

File: crates/astria-composer/tests/blackbox/helper/mod.rs [hunk 3]
```diff
@@ -20,9 +20,9 @@ pub mod mock_sequencer;
 static TELEMETRY: Lazy<()> = Lazy::new(|| {
     if std::env::var_os("TEST_LOG").is_some() {
         let filter_directives = std::env::var("RUST_LOG").unwrap_or_else(|_| "info".into());
-        telemetry::init(std::io::stdout, &filter_directives).unwrap();
+        telemetry::init(std::io::stdout, &filter_directives, None).unwrap();
```
[the remark this answers is not in the document]
Comment (joroshiba) [reply crates/astria-composer/tests/blackbox/helper/mod.rs:23]: The metrics crate mentions using `with_local_recorder` as a way to test, but I don't see any examples of what to use as the recorder. I don't know of a great way to test the prometheus exporter.

**Comment by joroshiba** [comment]
replaced w/ https://github.com/astriaorg/astria/pull/739

**Pull request closed without merging**